sky/packages/sky/lib/src/material/bottom_sheet.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+
+const Duration _kBottomSheetDuration = const Duration(milliseconds: 200);
+
+class _BottomSheet extends StatefulComponent {
+  _BottomSheet({
+    Key key,
+    this.child,
+    this.performance,
+    this.navigatorSize
+  }) : super(key: key);
+
+  final Widget child;
+  final PerformanceView performance;
+  final Size navigatorSize;
+
+  _BottomSheetState createState() => new _BottomSheetState();
+}
+
+class _BottomSheetState extends State<_BottomSheet> {
+
+  double _endHeight;
+
+  void _handleChildSizeChanged(Size newSize) {
+    final double maxSheetHeight = config.navigatorSize.height * 9.0 / 16.0;
+    setState(() {
+      _endHeight = math.min(maxSheetHeight, newSize.height);
+    });
+  }
+
+  Widget build(BuildContext context) {
+     return new SquashTransition(
+        performance: config.performance,
+        height: new AnimatedValue<double>(0.0, end: _endHeight ?? 0.0),
+        child: new ScrollableViewport(
```

Comment:
I'm assuming that using ScrollableViewport will only cause the sheet to be laid out once; that resizing the viewport will just cause a repaint. I took this approach, instead of just translating the sheet into view (as SnackBar does) so that the same _BottomSheet could be used for the persistent case. I tried to use ClipRect instead of ScrollableViewport since there isn't a plain Viewport and I don't want to enable scrolling, but I wasn't able to get that to work.